tests/test_red_list.py

```diff
@@ -0,0 +1,229 @@
+    user_id = "@alice:example"
+
+    async def test_other_data_type(self) -> None:
+        """Tests that incoming account data with a different account data type than the
+        one the module handles is ignored.
+        """
+        module, api = await create_module()
+
+        account_data_type = "org.matrix.foo"
+
+        await module.update_red_list_status(
+            user_id=self.user_id,
+            room_id=None,
+            account_data_type=account_data_type,
+            content={"foo": "bar"},
+        )
+
+        api.run_db_interaction.assert_not_called()
+        api.update_room_membership.assert_not_called()
+
+    async def test_no_hide_profile(self) -> None:
+        """Tests that account data of the right type but with a content that doesn't
+        include the hide_profile property is considered as if it was present and equal to
+        False.
+        """
+        module, api = await self._setup_user_in_list()
+
+        # Invalidate the cache, so it doesn't interfere with the call counts.
+        module._get_user_status.invalidate((self.user_id,))
+
+        # Trigger the callback with an account data that's missing the hide_profile key
+        # in its content.
+        await module.update_red_list_status(
+            user_id=self.user_id,
+            room_id=None,
+            account_data_type=ACCOUNT_DATA_TYPE,
+            content={},
+        )
+
+        # There should be two database calls: one to check the user's status, and one to
+        # update it.
+        self.assertEqual(
+            api.run_db_interaction.call_count, 2, api.run_db_interaction.mock_calls
+        )
+        api.update_room_membership.assert_not_called()
+
+        # Check that the user is not in the list anymore.
+        in_list, _ = await module._get_user_status(self.user_id)
+        self.assertFalse(in_list)
+
+    async def test_status_cached(self) -> None:
+        """Tests that users statuses are correctly cached and invalidated."""
+        module, api = await create_module()
+
+        # Get the user's status and check we made a database call.
+        in_list, _ = await module._get_user_status(self.user_id)
+        self.assertFalse(in_list)
+        api.run_db_interaction.assert_called_once()
+
+        # Get the user's status again and check we didn't make an additional database
+        # call.
+        in_list, _ = await module._get_user_status(self.user_id)
+        self.assertFalse(in_list)
+        api.run_db_interaction.assert_called_once()
+
+        # Add the user to the list and get their status again to check that it
+        # invalidated the cache (and caused another database call).
+        await module.update_red_list_status(
+            user_id=self.user_id,
+            room_id=None,
+            account_data_type=ACCOUNT_DATA_TYPE,
+            content={"hide_profile": True},
+        )
+        # Reset the mock so that the database call from update_red_list_status don't
```

Comment:
```suggestion
        # Reset the mock so that the database call from update_red_list_status doesn't
```
The same typo occurs a few lines below too.